File: sdks/java/lib/src/main/java/models/shared/ZipDownload.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * ZipDownload
 * Represents a successful request to create a `zip` archive of a list of files
 * and folders.
**/
public class ZipDownload {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("download_url")
    public String downloadUrl;
    public ZipDownload withDownloadUrl(String downloadUrl) {
        this.downloadUrl = downloadUrl;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("expires_at")
    public OffsetDateTime expiresAt;
    public ZipDownload withExpiresAt(OffsetDateTime expiresAt) {
        this.expiresAt = expiresAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name_conflicts")
    public ZipDownloadNameConflicts[][] nameConflicts;
    public ZipDownload withNameConflicts(ZipDownloadNameConflicts[][] nameConflicts) {
        this.nameConflicts = nameConflicts;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("status_url")
    public String statusUrl;
    public ZipDownload withStatusUrl(String statusUrl) {
        this.statusUrl = statusUrl;
        return this;
    }
}

File: sdks/go/pkg/models/shared/zipdownload.go
package shared

import (
	"time"
)

type ZipDownloadNameConflictsTypeEnum string

const (
	ZipDownloadNameConflictsTypeEnumFile   ZipDownloadNameConflictsTypeEnum = "file"
	ZipDownloadNameConflictsTypeEnumFolder ZipDownloadNameConflictsTypeEnum = "folder"
)

// ZipDownloadNameConflicts
// A file or folder for which a conflict was encountered, This object provides the type and identifier of the original item, as well as a mapping between its original name and it's new name as it will appear in the archive.
type ZipDownloadNameConflicts struct {
	DownloadName *string                           `json:"download_name,omitempty"`
	ID           *string                           `json:"id,omitempty"`
	OriginalName *string                           `json:"original_name,omitempty"`
	Type         *ZipDownloadNameConflictsTypeEnum `json:"type,omitempty"`
}

// ZipDownload
// Represents a successful request to create a `zip` archive of a list of files
// and folders.
type ZipDownload struct {
	DownloadURL   *string                      `json:"download_url,omitempty"`
	ExpiresAt     *time.Time                   `json:"expires_at,omitempty"`
	NameConflicts [][]ZipDownloadNameConflicts `json:"name_conflicts,omitempty"`
	StatusURL     *string                      `json:"status_url,omitempty"`
}
